From b2287ee49bdf9d76b1714f61591d6c2af4fbfb22 Mon Sep 17 00:00:00 2001
From: Shiva Kerdel <shiva@exdev.nl>
Date: Tue, 15 Nov 2016 09:30:03 +0100
Subject: [PATCH 071/107] Staging: fsl-mc: include: mc: Kernel type 'int'
 preferred over 's16'
Git-commit: cbafdc9816b4d59c678c08b7dbe4c1bce4a73553
Patch-mainline: v4.10-rc1
References: bsc#1041022

After following a discussion about the used integer types Dan Carpenter
pointed out that 'int' types should be used over the current change to
's16'. The reason for this is to have an upper bound instead of overflowing
the 's16' so we could still remove devices.

Signed-off-by: Shiva Kerdel <shiva@exdev.nl>
Suggested-by: Stuart Yoder <stuart.yoder@nxp.com>
Suggested-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Alexander Graf <agraf@suse.de>

---
 drivers/staging/fsl-mc/include/mc-bus.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl-mc/include/mc-bus.h b/drivers/staging/fsl-mc/include/mc-bus.h
index c7cad87..42700de 100644
--- a/drivers/staging/fsl-mc/include/mc-bus.h
+++ b/drivers/staging/fsl-mc/include/mc-bus.h
@@ -42,8 +42,8 @@ struct msi_domain_info;
  */
 struct fsl_mc_resource_pool {
 	enum fsl_mc_pool_type type;
-	s16 max_count;
-	s16 free_count;
+	int max_count;
+	int free_count;
 	struct mutex mutex;	/* serializes access to free_list */
 	struct list_head free_list;
 	struct fsl_mc_bus *mc_bus;
-- 
1.8.5.6

